contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/utils/HadoopUtils.java

```diff
@@ -326,10 +334,34 @@ private static Configuration getConfiguration(VoldemortConfig voldemortConfig, S
         hadoopConfig.set("hadoop.security.group.mapping",
                    "org.apache.hadoop.security.ShellBasedUnixGroupsMapping");
 
+        // sourceScheme is null for file based paths, used in testing.
+        if(sourceScheme != null) {
+            /*
+             * Hadoop FileSystem class caches the FileSystem objects based on
+             * the scheme , authority and UserGroupInformation.
+             * 
+             * The default config was to generate new UserGroupInformation for
```

Comment:
Yes we do. I tried implementing the re-use of UGI instances in the past, but it was unstable and I didn't have time to fully debug it, so I made it optional and hid it behind a config. The config's JavaDoc does have a warning that the feature is unstable.